Client/src/main/java/org/projectcardboard/client/view/battleview/TroopAnimation.java

```diff
@@ -250,7 +249,8 @@ private void setAction(ACTION action) {
       imageView.setScaleX(1);
     else
       imageView.setScaleX(-1);
-    switch (action) {
+    switch (action) { // Todo check if order matters. E.g. is death before attack the cause of our
```

Comment:
Order does matter in Java switch statements. Since `DEATH` is before `ATTACK` and it `breaks`, it'll find it first and do it. I don't _think_ this is the cause of that bug though? `setAction(ACTION.DEATH)` happens if `shouldDie == true`, which only happens if `kill()` has been called... the bug is likely that we're somehow setting `shouldDie = true` or `ACTION.DEATH` before we've set or processed `ACTION.ATTACK`.

Reply:
yes, tested this in a different branch. This isn't the cause of the dying animation bug